### PR TITLE
ARROW-16873: [Python] Disable faulthandler on spawned child subprocess on run_debug_memory_pool tests

### PR DESCRIPTION
**Pull request by raulcd**

### CODE REVIEW

**Comment by pitrou** [comment]
Can you print the value of `faulthandler.is_enabled()` in the child process @raulcd ?

**Comment by raulcd** [comment]
It seems is enabled:
```
_____________ test_debug_memory_pool_disabled[default_memory_pool] _____________
----------------------------- Captured stdout call -----------------------------
faulthandler.is_enabled():  True
----------------------------- Captured stderr call -----------------------------
____________ test_debug_memory_pool_disabled[jemalloc_memory_pool] _____________
----------------------------- Captured stdout call -----------------------------
faulthandler.is_enabled():  True
----------------------------- Captured stderr call -----------------------------
```
What memory alocator is supposed to be the `system_memory_pool`?
```
tests/test_memory.py::test_debug_memory_pool_disabled[default_memory_pool] PASSED [ 51%]
tests/test_memory.py::test_debug_memory_pool_disabled[jemalloc_memory_pool] PASSED [ 51%]
tests/test_memory.py::test_debug_memory_pool_disabled[system_memory_pool] FAILED [ 51%]
```
I am going to try a couple of things to try and disable `faulthandler`

File: ci/scripts/python_test.sh [hunk 1]
```diff
@@ -54,4 +54,4 @@ export PYARROW_TEST_ORC
 export PYARROW_TEST_PARQUET
 export PYARROW_TEST_S3
 
-pytest -r s -v ${PYTEST_ARGS} --pyargs pyarrow
+pytest -r P -v ${PYTEST_ARGS} --pyargs pyarrow
```
Comment (pitrou) [inline ci/scripts/python_test.sh:57]: Hmm, why the change? Should we undo it?

File: python/pyarrow/tests/test_memory.py [hunk 2]
```diff
@@ -188,6 +188,10 @@ def run_debug_memory_pool(pool_factory, env_value):
     code = f"""if 1:
         import ctypes
         import pyarrow as pa
+        import faulthandler
+        if faulthandler.is_enabled():
+            faulthandler.disable()
+        print("faulthandler.is_enabled(): ", faulthandler.is_enabled())
```
Comment (pitrou) [inline python/pyarrow/tests/test_memory.py:194]:
```suggestion
        # ARROW-16873: some Python installs enable faulthandler by default,
        # which could dump a spurious stack trace if the following crashes
        import faulthandler
        faulthandler.disable()
```

File: ci/scripts/python_test.sh (hunk 1 above)
Comment (raulcd) [reply ci/scripts/python_test.sh:57]: yes, this was only temporary to be able to see the print output on CI. I'll revert it back.

Comment (raulcd) [reply ci/scripts/python_test.sh]: removed

**Comment by pitrou** [comment]
@github-actions crossbow submit -g python

**Comment by pitrou** [comment]
@github-actions crossbow submit -g python

**Comment by pitrou** [comment]
(I closed this PR by mistake, sorry)

**Comment by pitrou** [comment]
Ok, I understand: it seems that "Close with comment" is now the default choice in the Github UI when pressing Ctrl+Enter?

**Review by kou — APPROVED** [review]
+1
